src/Features/CSharp/Portable/Snippets/CSharpIfSnippetProvider.cs

```diff
@@ -38,7 +31,49 @@ protected override void GetIfStatementConditionAndCursorPosition(SyntaxNode node
         {
             var ifStatement = (IfStatementSyntax)node;
             condition = ifStatement.Condition;
-            cursorPositionNode = ifStatement.Statement.SpanStart + 1;
+            var blockStatement = (BlockSyntax)ifStatement.Statement;
+
+            var triviaSpan = blockStatement.CloseBraceToken.LeadingTrivia.Span;
+
+            // Getting the midpoint location of the trivia to find where the cursor shoudl end up
+            cursorPositionNode = triviaSpan.Start + (triviaSpan.Length / 2) + 1;
```

Comment:
`cursorPositionNode` is a misleading name here. 

Also, why is it `/2` here? What makes the mid point special?

Reply:
elaborated in the comment